src/games/automatchframe.py

```diff
@@ -0,0 +1,218 @@
+
+
+class MatchmakerQueue(FormClass, BaseClass):
+
+    def __init__(self, games, client, queueName, teamSize):
+        BaseClass.__init__(self, games)
+        self.setupUi(self)
+
+        self.queueName = queueName
+        self.teamSize = teamSize
+        self.subFactions = Settings.get(
+            "play/{}Factions".format(self.queueName),
+            default=[False] * 4,
+            type=bool
+        )
+        self.games = games
+        self.client = client
+        self.client.matchmaker_info.connect(self.handleQueueInfo)
+        self.games.matchmaker_search_info.connect(self.handleSearchInfo)
+        self.games.match_found_message.connect(self.handleMatchFound)
+        self.games.stop_search_ranked_game.connect(self.stopSearchRanked)
+        self.games.party_updated.connect(self.handlePartyUpdate)
+
+        self._rankedIcons = {
+            Factions.AEON: self.rankedAeon,
+            Factions.CYBRAN: self.rankedCybran,
+            Factions.SERAPHIM: self.rankedSeraphim,
+            Factions.UEF: self.rankedUEF,
+        }
+        self.rankedUEF.setIcon(util.THEME.icon("games/automatch/uef.png"))
+        self.rankedAeon.setIcon(util.THEME.icon("games/automatch/aeon.png"))
+        self.rankedCybran.setIcon(
+            util.THEME.icon("games/automatch/cybran.png")
+        )
+        self.rankedSeraphim.setIcon(
+            util.THEME.icon("games/automatch/seraphim.png")
+        )
+
+        self.searching = False
+        self.updatePlayButton()
+
+        self.rankedPlay.clicked.connect(self.startSearchRanked)
+        self.rankedPlay.show()
+        self.mapsPool.clicked.connect(self.startViewMapsPool)
+
+        self.setFactionIcons(self.subFactions)
+
+        keys = (
+            QtCore.Qt.Key_1, QtCore.Qt.Key_2, QtCore.Qt.Key_3, QtCore.Qt.Key_4
+        )
+        self.shortcut = QtWidgets.QShortcut(
+            QtGui.QKeySequence(QtCore.Qt.CTRL + keys[self.teamSize-1]),
+            self.client,
+            self.startSearchRanked
+        )
+        self.games.matchmakerShortcuts.append(self.shortcut)
+
+        self.matchmakerTimer = QtCore.QTimer()
+        self.matchmakerTimer.timeout.connect(self.updateMatchmakerTimer)
+        self.secondsToAutomatch = 0
+
+    def setFactionIcons(self, subFactions):
+        for faction, icon in self._rankedIcons.items():
+            try:
+                icon.clicked.disconnect()
+            except TypeError:
+                pass
+            icon.setChecked(subFactions[faction.value-1])
+            icon.clicked.connect(
+                partial(self.selectFaction, factionID=faction.value)
+            )
+
+    def handleQueueInfo(self, message):
+        for queue in message.get("queues", {}):
+            if queue["queue_name"] == self.queueName:
+                self.labelInQueue.setText(
+                    "In Queue: {}".format(queue["num_players"])
+                )
+                self.secondsToAutomatch = int(queue["queue_pop_time_delta"])
```

Comment:
O.o

Reply:
sorry, can't understand you. probably it would be better to update all text labels in one function, but i didn't want to update another label every second.
using `int` i think is better, because i would prefer to stare for almost another second at `00:00` than see match found message at `00:01`

Reply:
Oh, it's just I think the java client doesn't even use this field yet since I added it after `queue_pop_time` was sometimes unreliable when people had weird system times. But technically `queue_pop_time_delta` will be a bit off depending on network latency.